### DISCUSSION TITLE
disable process.on('SIGTERM') SIGQUIT SIGINT

### DISCUSSION THREAD

**Discussion by inkeytech** [QuorumDMS/ftp-srv#246] [Q&A] [answered]
Hello,

Would it be possible to include an option to disable the "quit" handlers of ftp-srv?

process.on('SIGTERM', quit);
process.on('SIGINT', quit);
process.on('SIGQUIT', quit);

I would like my app, using the excellent library, to be able to gracefully close the FTP server IN ADDITION TO closing other connections and processes.

With the ftp-srv event listeners in place, my own handlers fail to function.

I'm currently circumventing this by commenting out the 3 lines above, but this is not a sustainable solution.

Any pointers would be much appreciated.

Thanks,

Ian

**Comment by trs** [comment]
This is a great point. From a library perspective it doesn't make a lot of sense to hijack the process and end it like that.
Do you think keeping the `SIG-` events to call `close` would make more sense? Or just remove them completely? Either way would be a breaking change.
A boolean option could be added to the [constructor](https://github.com/autovance/ftp-srv/blob/master/src/index.js#L13) that would only call each `process.on` if true. This wouldn't be breaking and would satisfy your requirements.

```js
class FtpServer extends EventEmitter {
  constructor(options = {}) {
    super();
    this.options = Object.assign({
      ...,
      endProcessSig: true // Better name?
    }, options);

    ...

    if (this.options.endProcessSig) {
      process.on('SIGTERM', quit);
      process.on('SIGINT', quit);
      process.on('SIGQUIT', quit);
    }
  }
}
```

**Reply by matt-forster** [reply]
`endOnProcessSignal` maybe?

**Comment by inkeytech** [comment]
Hello Tyler,

Thank you for the prompt and informative response. And compliments on 
your excellent library.

I would not want you to include any breaking changes.

And having the current default behaviour of automatically calling 
process.exit() is not an entirely bad idea.

But, I'd definitely be in favour of you including the changes you 
detailed that extend EventEmitter to check for the state of a newly 
provided boolean option.

That looks like a neat and intuitive approach.

Thanks again,

Ian



------ Original Message ------

**Comment by matt-forster** [comment, marked as answer]
I've converted this discussion to a feature request here: #280
